lib/tile_bag.rb

```diff
@@ -0,0 +1,63 @@
+class TileBag
+  attr_accessor :tile_hash, :tile_bag
+  def initialize
+    @tile_hash = {
```

Comment:
Since this hash is the same for every instance of `TileBag` you might consider making it a constant rather than an instance variable. At that point you wouldn't need the `attr_accessor :tile_hash` part, either.